mir_eval/transcription.py

```diff
@@ -353,7 +523,182 @@ def precision_recall_f1(ref_intervals, ref_pitches, est_intervals, est_pitches,
     precision = float(len(matching))/len(est_pitches)
     recall = float(len(matching))/len(ref_pitches)
     f_measure = util.f_measure(precision, recall)
-    return precision, recall, f_measure
+
+    avg_overlap_ratio = average_overlap_ratio(ref_intervals, est_intervals,
+                                              matching)
+
+    return precision, recall, f_measure, avg_overlap_ratio
+
+
+def average_overlap_ratio(ref_intervals, est_intervals, matching):
+    """Compute the Average Overlap Ratio between a reference and estimate
+    note transcription. Given a reference and corresponding estimate note,
+    their overlap ratio (OR) is defined as the ratio between the duration of
+    the time segment in which the two notes overlap and the time segment
+    spanned by the two notes combined (earliest onset to latest offset): OR =
+    (min(ref_offset, est_offset) - max(ref_onset, est_onset)) /
+    (max(ref_offset, est_offset) - min(ref_onset, est_onset)). The Average
+    Overlap Ratio (AOR) is given the mean OR computed over all matching
+    reference and estimate notes. The metric goes from 0 (worst) to 1 (best).
+
+    Note: this function assumes the matching of reference and estimate notes
+    (see :func:`match_notes`) has already been performed and is provided by the
+    `matching` parameter.
+
+    Parameters
+    ----------
+    ref_intervals : np.ndarray, shape=(n,2)
+        Array of reference notes time intervals (onset and offset times)
+    est_intervals : np.ndarray, shape=(m,2)
+        Array of estimated notes time intervals (onset and offset times)
+    matching : list of tuples
+        A list of matched reference and estimated notes.
+        ``matching[i] == (i, j)`` where reference note i matches estimate note
+        j.
+
+    Returns
+    -------
+    avg_overlap_ratio: float
+        The computed Average Overlap Ratio score
+    """
+    ratios = []
+    for match in matching:
+        ref_int = ref_intervals[match[0]]
+        est_int = est_intervals[match[1]]
+        overlap_ratio = (
+            (min(ref_int[1], est_int[1]) - max(ref_int[0], est_int[0])) /
+            (max(ref_int[1], est_int[1]) - min(ref_int[0], est_int[0])))
```

Comment:
For the denominator to be 0 you need both intervals to have exactly the same start time *and* a duration of zero, which should get caught during validation (assuming your calling this function via `precision_recall_f1_overlap`). However:

1. The existing `validate` function in `transcription.py` doesn't catch this even though it should. I'll update `validate` to call `util.validate_intervals` internally, which will fix this.
2. If for some reason someone calls this function manually without validation then yes, it's not impossible for the denominator to evaluate to 0 (because the input data is invalid). Is it worth adding a check for this?